examples/checkout-book.yaml

```diff
@@ -0,0 +1,93 @@
+'@context':
+  - https://git.io/vf-examples-jsonld-context
+  - library: https://library.example/
+    bob: https://bob.example/
+
+
+'@id': rgh:valueflows/valueflows/master/examples/checkout-book.yaml
+'@graph':
+
+  # resource before
+  - '@id': library:2402b57c-1c45-423a-acb2-f50961c97d65
+    '@type': EconomicResource
+    skos:note: Valueflows cookbook
+    currentQuantity:
+      qudt:unit: unit:Number
+      qudt:numericValue: 2
+
+  - '@id': library:21f361a6-2375-46bb-b192-c21b5ba833bf
+    '@type': EconomicResource
+    skos:note: Possession rights and responsibilities for the Valueflows cookbook
+    underlyingResource: library:2402b57c-1c45-423a-acb2-f50961c97d65 # maybe needs different name
+    currentQuantity:
+      qudt:unit: unit:Number
+      qudt:numericValue: 2
+
+  # check out
+  - '@id': library:f47064eb-7120-4b31-b882-770165901fe6
+    '@type': Process
+    skos:note: renting Valueflows cookbook to Bob
+
+  - '@id': library:54b814ee-62dc-40c1-bb96-f8582aa4f771
+    '@type': EconomicEvent
+    inputOf: library:f47064eb-7120-4b31-b882-770165901fe6
+    action: give
+    resourceInventoriedAs: library:21f361a6-2375-46bb-b192-c21b5ba833bf #library's possession resource
```

Comment:
since we have *give* on possession 'resource' we don't have any action on 'underlying resource' so if underlying resource has current physical location, and current quantity 2, it looks like both books still stay at the library.

How do we handle situation when library transfers rights to possess book on monday and bob doesn't excercise those rights until wednesday. So for two days Bob already has rights but underlying book still stays at the library, on wednesday we have another event where Bob picks up 1 copy of the book but we don't have transfer o rights any more since he had rights to do it already on monday.

Reply:
> if underlying resource has current physical location

Yeah I ran into that in the shipping example I'm working on.  I didn't figure the library needed a location defined.  But still it is a valid question.  In the shipping example, I'm going to go back to an issue/receive to a new resource for the stuff that will go on the truck, which I initially removed.

>How do we handle situation when library transfers rights to possess book on monday and bob doesn't excercise those rights until wednesday

I'll give that a try.  I think that is an 'availability' right, since nobody else can check out the book in the meantime, it is reserved.

Reply:
If we talk about rights, having rights to posses doesn't mean that one possesses something already, it just means that one can possess something while having those rights. Does your 'rights' imply actual possession or just rights to posses (which one may exercise at some point or not)

Reply:
Very true.  I pulled possession out of the rights at one point, probably need to do it again. 

And it seems like I need to connect the underlying resource and the rights more tightly in some way, something more like a bundle than an underlying resource.  You do need to have possession of some sort in order to for example use something.